companion/lib/Surface/Group.ts

```diff
@@ -248,39 +250,85 @@ export class SurfaceGroup {
 		this.#pageHistory.index = 0
 	}
 
+	/**
+	 * Get first page ID, taking into account `groupConfig.allowed_page_ids`
+	 */
+	#getFirstPageId(): string {
+		const validPages = this.groupConfig.restrict_pages ? (this.groupConfig.allowed_page_ids ?? []) : []
+		return validPages.length > 0 ? validPages[0] : this.#pageStore.getFirstPageId()
```

Comment:
Perhaps this should be checking that the contents of `validPages` are valid according to `this.#pageStore.isPageIdValid(pageID)`?